### PR TITLE
fix(cosyvoice): repair installation and preserve generated speech context

### PR DESCRIPTION
**Pull request by debpalash**
CosyVoice could report a successful installation, then fail on missing imports or generate unrelated/truncated speech. This repairs the managed runtime and its compatibility with current Transformers.

- Restore gdown, wget and pyarrow. Build pinned upstream PyWORLD/WORLD sources that use importlib.metadata rather than downgrading setuptools. Verify dataset and Matcha imports before marking installation complete.
- Reject obsolete managed runtime markers while retaining model weights and leaving user-managed environments intact. Document the native compiler requirement.
- Load Qwen in float32 before applying trained weights, preserving explicit upstream precision choices. Complete incremental attention masks with cached prompt tokens so newer Transformers retains the text and voice context. Keep legacy cache support and existing full masks.

Validation: 190 targeted installer, sidecar and changelog checks pass offline with an empty HF cache. The tiny real-Qwen cached/full-context regression fails without the mask fix and passes with it. PyWORLD native pitch extraction passes in isolated Python 3.10 environments on Windows, macOS ARM and Linux. A clean managed Linux CPU install and real CosyVoice 3 generation were exercised: the corrected 89-word English passage is recovered by an independent local speech recognizer, whereas the previous runtime produced no matching transcript. This does not claim real-model GPU or every-language coverage.

Related #2096. Full CI and review must pass before landing.

Restored CosyVoice runtime dependencies and pinned PyWORLD/WORLD sources, with managed-install marker checks that preserve legacy and user-managed environments. Added Transformers 5 Qwen compatibility for float32 checkpoint loading and cached attention masks to restore generated speech context. Validation covered isolated native builds, managed Linux CPU installation, and real CosyVoice 3 generation; GPU and all-language coverage remain unverified.

### CODE REVIEW

**Comment by debpalash** [comment]
Additional validation: the complete CosyVoice requirements resolve with Python 3.10, torch 2.7.0 and torchaudio 2.7.0 in the isolated environment. All restored imports and native PyWORLD pitch extraction pass without pkg_resources; cross-platform build checks are running.

**Comment by debpalash** [comment]
The isolated Python 3.10 dependency smoke is now green on macOS ARM, Windows and Linux (run 35517516087). Each platform builds the pinned native PyWORLD source, imports all restored dependencies without pkg_resources, and executes pitch extraction. The initial Linux/Windows smoke failure was inherited runner site-packages; an explicit isolated venv fixed the test setup. Full managed CPU installation is now being exercised separately.

File: scripts/smoke-cosyvoice-dependencies.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""Check restored native/runtime dependencies in the isolated Python 3.10 env."""
+import importlib.util
+import gdown
```
[the remark this answers is not in the document]
Comment (debpalash) [reply scripts/smoke-cosyvoice-dependencies.py:3]: Intentional import smoke: executing this import is the check. This script verifies the missing-at-runtime failure reported in #2096; it must not invoke the network download function.

File: scripts/smoke-cosyvoice-dependencies.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import numpy as np
+import pyarrow.parquet
```
[the remark this answers is not in the document]
Comment (debpalash) [reply scripts/smoke-cosyvoice-dependencies.py:5]: Intentional import smoke: importing the native parquet module exercises the late dependency that the old installer missed.

File: scripts/smoke-cosyvoice-dependencies.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import pyworld
+import wget
```
[the remark this answers is not in the document]
Comment (debpalash) [reply scripts/smoke-cosyvoice-dependencies.py:7]: Intentional import smoke: executing this import is the runtime compatibility check; downloading a file would add an unrelated network dependency to this test.

File: backend/services/sidecar_install.py [hunk 4]
```diff
@@ -561,9 +562,22 @@ def probe() -> bool:
...
+                path="third_party/PyWorld/lib/World",
+                revision="d625e7608ca23a870018f01e7c562ac683d9847f",
+                tarball_url="https://github.com/mmorise/World/archive/d625e7608ca23a870018f01e7c562ac683d9847f.tar.gz",
+                required_path="src/dio.cpp",
+            ),
         ),
         venv_args=("--python", "3.10"),
-        install_args=("-r", _COSYVOICE_REQUIREMENTS),
+        install_args=("-r", _COSYVOICE_REQUIREMENTS, "{checkout}/third_party/PyWorld"),
+        install_revision="inference-imports-v2",
```
[the remark this answers is not in the document]
Comment (debpalash) [reply backend/services/sidecar_install.py:580]: Fixed in 87c5097d. Managed runtime selection and installer health now share the recipe-marker verdict; stale managed environments cannot be selected. User-managed environments retain their existing contract. Regression covers CosyVoice and the other revisioned recipe (MOSS Nano), including external installs.

File: backend/engines/cosyvoice_subprocess/main.py [hunk 5]
```diff
@@ -136,6 +136,67 @@ def _model_dir(checkout: str) -> str:
...
+    return torch.cat((prefix, masks), dim=-1)
+
+
+def _repair_qwen_cache(model):
+    from cosyvoice.llm.llm import Qwen2Encoder
+
+    components = getattr(model, "model", model)
+    encoder = getattr(getattr(components, "llm", None), "llm", None)
+    if not isinstance(encoder, Qwen2Encoder):
+        return  # CosyVoice 1 uses a different encoder.
```
[the remark this answers is not in the document]
Comment (debpalash) [reply backend/engines/cosyvoice_subprocess/main.py]: Fixed on the original branch: both Qwen symbols are optional for legacy non-Qwen checkouts. The new full-loader regression fails before the fix and passes after it; all 23 sidecar tests pass with an empty offline HF cache.

**Comment by debpalash** [comment]
Additional real-model validation of the current fix: CosyVoice 3 generated the complete Spanish advertisement passage in one call (27.12 s audio, 86.8 s CPU generation), using the installed Transformers 5.10.1 runtime. After the normal synthetic-audio marker, independent local Whisper tiny multilingual recovered every sentence through the final phrase; it made several word/name substitutions, so this is evidence against the old truncation/lost-context failure, not a claim of perfect pronunciation or exhaustive language/GPU coverage. Earlier English validation recovered all 89 words. No external speech service was used.
